api/agent/lb_agent.go

```diff
@@ -179,6 +190,45 @@ func (a *lbAgent) Submit(callI Call) error {
 	return a.handleCallEnd(ctx, call, err, true)
 }
 
+// setRequestGetBody sets GetBody function on the given http.Request if it is missing.  GetBody allows
+// reading from the request body without mutating the state of the request.
+func (a *lbAgent) setRequestBody(ctx context.Context, call *call) error {
+
+	r := call.req
+	if r.Body == nil || r.GetBody != nil {
+		return nil
+	}
+
+	// WARNING: we need to handle IO in a separate go-routine below
+	// to be able to detect a ctx timeout. When we timeout, we
+	// let gin/http-server to unblock the go-routine below.
+	errApp := make(chan error, 1)
+	go func() {
+
+		buffer, err := ioutil.ReadAll(r.Body)
```

Comment:
Mentioned this in a prior review, but is an unbounded ReadAll safe here?  Is there something else enforcing the maximum size of the request body, or do we need to do it ourselves?